library/std_detect/src/detect/os/riscv.rs

```diff
@@ -119,11 +119,31 @@ pub(crate) fn imply_features(mut value: cache::Initializer) -> cache::Initialize
         imply!(d | zfhmin | zfa => f);
         imply!(zfbfmin => f); // and some of (not all) "Zfh" instructions.
 
-        // Relatively complex implication rules from the "C" extension.
+        // Relatively complex implication rules around the "C" extension.
+        // (from "C" and some others)
         imply!(c => zca);
         imply!(c & d => zcd);
         #[cfg(target_arch = "riscv32")]
         imply!(c & f => zcf);
+        // (to "C"; defined as superset)
+        cfg_select! {
+            target_arch = "riscv32" => {
+                if value.test(Feature::d as u32) {
+                    imply!(zcf & zcd => c);
```

Comment:
Shouldn't this be `zcf & zcd & zca => c`? Or am I misunderstanding the implication here?

Reply:
You are correct but this is redundant as `zca` is implied by both `zcd` and `zcf`.

cf. https://github.com/rust-lang/rust/blob/ee7627ee409b1f2a3ebbf6c868adbc0d75a1f43b/library/std_detect/src/detect/os/riscv.rs#L98-L99